From 2eff60e37f55d6870e014a0652b7fc9ca818de04 Mon Sep 17 00:00:00 2001
From: Ajay Kumar Gupta <ajay.gupta@ti.com>
Date: Mon, 27 Feb 2012 15:12:34 +0530
Subject: [PATCH 160/609] ARM: OMAP: AM33XX: Cleanup usb hwmod

Original commit: e209178fab09b5d56fb7b236770a876aa1b3cd76

  From git://arago-project.org/git/projects/linux-am33x.git
  And ti-sdk-am335x-evm-05.05.00.00-Linux-x86-Install image

Updated the hwmod data for usb pm support.
Changes:
	- updated the sysc type to type2
	- Added the interface and functional clock and removed opt clocks

Signed-off-by: Ajay Kumar Gupta <ajay.gupta@ti.com>
---
 arch/arm/mach-omap2/omap_hwmod_33xx_data.c |   11 +++--------
 1 files changed, 3 insertions(+), 8 deletions(-)

diff --git a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
index cbb1983..47d7e16 100644
--- a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
+++ b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
@@ -2913,6 +2913,7 @@ static struct omap_hwmod_irq_info am33xx_usbss_mpu_irqs[] = {
 static struct omap_hwmod_ocp_if am33xx_l3_slow__usbss = {
 	.master		= &am33xx_l3slow_hwmod,
 	.slave		= &am33xx_usbss_hwmod,
+	.clk		= "usbotg_ick",
 	.addr		= am33xx_usbss_addr_space,
 	.user		= OCP_USER_MPU,
 	.flags		= OCPIF_SWSUP_IDLE,
@@ -2922,16 +2923,12 @@ static struct omap_hwmod_ocp_if *am33xx_usbss_slaves[] = {
 	&am33xx_l3_slow__usbss,
 };
 
-static struct omap_hwmod_opt_clk usbss_opt_clks[] = {
-	{ .role = "clkdcoldo", .clk = "usbotg_fck" },
-};
-
 static struct omap_hwmod am33xx_usbss_hwmod = {
 	.name		= "usb_otg_hs",
 	.class		= &am33xx_usbotg_class,
-	.clkdm_name	= "l4ls_clkdm",
+	.clkdm_name	= "wkup_usb_clkdm",
 	.mpu_irqs	= am33xx_usbss_mpu_irqs,
-	.main_clk	= "usbotg_ick",
+	.main_clk	= "usbotg_fck",
 	.flags		= HWMOD_CONTROL_OPT_CLKS_IN_RESET,
 	.prcm		= {
 		.omap4	= {
@@ -2939,8 +2936,6 @@ static struct omap_hwmod am33xx_usbss_hwmod = {
 			.modulemode	= MODULEMODE_SWCTRL,
 		},
 	},
-	.opt_clks	= usbss_opt_clks,
-	.opt_clks_cnt	= ARRAY_SIZE(usbss_opt_clks),
 	.slaves		= am33xx_usbss_slaves,
 	.slaves_cnt	= ARRAY_SIZE(am33xx_usbss_slaves),
 };
-- 
1.7.5.4

